powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/PowerAuthService.java

```diff
@@ -1156,6 +1175,31 @@ public GetEciesDecryptorResponse getEciesDecryptor(GetEciesDecryptorRequest requ
         }
     }
 
+    @Transactional
+    public GetEciesEncryptorResponse getEciesEncryptor(GetEciesEncryptorRequest request) throws GenericServiceException {
+        if (request.getApplicationKey() == null || request.getEphemeralPublicKey() == null) {
+            logger.warn("Invalid request parameters in method getEciesEncryptor");
+            // Rollback is not required, database is not used for writing
+            throw localizationProvider.buildExceptionForCode(ServiceError.INVALID_REQUEST);
+        }
+        // The activationId value can be null in case the decryptor is used in application scope
+        try {
+            logger.info("GetEciesEncryptorRequest received, application key: {}, activation ID: {}", request.getApplicationKey(), request.getActivationId());
+            final GetEciesEncryptorResponse response = behavior.getEciesEncryptionBehavior().getEciesEncryptorParameters(request);
+            logger.info("GetEciesEncryptorRequest succeeded");
+            return response;
+        } catch (GenericServiceException ex) {
+            // already logged
+            throw ex;
+        } catch (RuntimeException | Error ex) {
```

Comment:
We should not catch `Error`. The fix will be done in a different PR but I suggest not to copy&paste this pattern to the new code. See https://rules.sonarsource.com/java/RSPEC-1181/

Reply:
Let's raise a separate issue.